src/magql/resolver_factory.py

```diff
@@ -529,6 +529,9 @@ class SingleResolver(QueryResolver):
     the instance specified by id.
     """
 
+    def post_resolve(self, resolved_value, parent, info, *args, **kwargs):
+        return resolved_value if resolved_value else {"result": resolved_value}
```

Comment:
We have had the design discussion to remove the result dictionary but here it is not possible (at least with the current setup). Take an example with a `Threat` model. From the top level query, the `threat` query returns a non-nullable Payload object. (The result and/or error key of the Payload object could be null). The resolver for the `threat` query is the `single_resolver`. In the case that there is no associated id, the `single_resolver` returns `None`. Upon completing this `None` value, GraphQL raises an error because `None` has been returned in a non-nullable field. The work around is to not return a `None` value. This isn't a problem for `many_query` because it returns an empty list, which passes the null check. Looking into this for update and delete, they also return a GQL error that no row was found with that ID. I think dealing with the mutations is another topic though.


Here is the GraphQL null check for reference:
```
def is_nullish(value: Any) -> bool:
    """Return true if a value is null, undefined, or NaN."""
    return value is None or value is INVALID or value != value
```

Reply:
OK, that makes sense.